### ISSUE TITLE
FieldMissingNullable does not recognize @MonotonicNonNull

### ISSUE THREAD

**Issue by imoverclocked** [google/error-prone#2880]
```java
import org.checkerframework.checker.nullness.qual.MonotonicNonNull;

// ...

@MonotonicNonNull String lazySetting;

String getSetting() {
    if (lazySetting == null) { // This line triggers FieldMissingNullable in errorprone 2.10.0
        lazySetting = "foo";
    }
    return lazySetting;
}
```

While MonotonicNonNull is part of the checker framework, it is recognized by other nullability checks (eg: NullAway)

**Comment by cpovirk** [comment]
Thanks for the report. We got a report of this internally, too, and we fixed it, but it came too late for the current release. The fix will be in the next release: https://github.com/google/error-prone/commit/724252f3cf3dadff014dc232eb696c42722c3a0f

**Comment by imoverclocked** [comment]
In the commit message:

> In practice, I think this is a reasonable approach: `@MonotonicNonNull` is likely to be used only in CF-checked code, so it's fairly likely to be correct already.

Just FYI: This can also be recognized/enforced with NullAway. Assigning a nullable value to a `@MonotonicNonNull` type will be recognized as a problem.

**Comment by cpovirk** [comment]
Thanks, good to know.

For anyone else reading this: I don't actually know whether most tools recognize MonotonicNonNull. Google's primary nullness checkers are the Checker Framework and Kotlin, so the lack of support I was referring to was Kotlin's.
